Upgrade to Span V2

llinder opened the pull request:
Upgrades things to the Span V2 model.

Also started pushing the trace ID into the MDC so its easily available for log correlation.

llinder commented on src/main/java/ratpack/zipkin/internal/RatpackCurrentTraceContext.java:
```diff
@@ -21,26 +25,47 @@ public RatpackCurrentTraceContext() {
-  public static final class TraceContextHolder {
+  private static final class ContextHolder {
+
+    private static final ContextHolder EMPTY = new ContextHolder(null);
```
An empty value to avoid extra object creation every time the last Scope is closed.

llinder commented on src/main/java/ratpack/zipkin/internal/RatpackCurrentTraceContext.java:
```diff
@@ -21,26 +25,47 @@ public RatpackCurrentTraceContext() {
+      return () -> {
+        if (previous.isPresent() && previous.get().context != null) {
+          registrySupplier.get().add(previous);
+          MDC.put(MCD_TRACEID_KEY, previous.get().context.traceIdString());
```
We could probably make the MDC stuff optional if others feel its not always desirable.  I thought I would just throw it in here and get feedback first before doing extra work to make it optional :)

hyleung replied:
@llinder personally, I'm a fan of MDC, so I'm not opposed to having it in. But if it's not too tied into the upgrade to V2 spans, you could just split it out into a separate PR and we can work on getting this one merged before tackling that.

llinder replied:
Sure I will make it a different PR.  Actually that part is what got me started on this because I need something like that sooner than the V2 stuff.  Figured I would try and tackle the V2 stuff as long as I was in the code though.

llinder commented on src/test/groovy/ratpack/zipkin/ServerTracingModuleSpec.groovy:
```diff
@@ -55,7 +53,7 @@ class ServerTracingModuleSpec extends Specification {
 								.clientSampler(HttpSampler.TRACE_ID)
 								.serverSampler(HttpSampler.TRACE_ID)
-								.spanReporter(Reporter.NOOP)
+								.spanReporter({Span s -> }) // noop
```
Reporter.NOOP uses Span V1.  Not sure if there is an equivalent type for Span V2 but I couldn't find one.

codefromthecrypt replied:
there isn't at the moment, we could add NOOP2? :)

llinder commented on src/main/java/ratpack/zipkin/TracedParallelBatch.java:
```diff
@@ -60,8 +60,7 @@ private TracedParallelBatch(Iterable<? extends Promise<T>> promises) {
     return ParallelBatch.of(promises)
-        .execInit(execution ->
-            execution.add(new RatpackCurrentTraceContext.TraceContextHolder(context)));
+        .execInit(execution -> execution.add(RatpackCurrentTraceContext.wrap(context)));
```
Really wish this wasn't referring to `RatpackCurrentTraceContext` directly here.  It basically makes it impossible to use a different implementation.

Long term I think this API probably needs some deeper thought.  For now I'm just punting on it but should probably make a ticket and try to come up with a better solution.

hyleung replied:
@llinder by "different implementation", do you mean a different implementation of `CurrentTraceContext`?

llinder replied:
Yeah, meant different implementation of CurrentTraceContext.  Right now this is tightly coupled to RatpackCurrentTraceContext which in hind sight feels less than ideal. :(

hyleung replied:
@llinder what use case can you think of where we'd require a different implementation of `CurrentTraceConext`? In any case, as long as this stuff is an implementation detail of (in this case `TraceParallelBatch` and it doesn't leak out, I figure we're free to change it later if we feel it's necessary.

llinder replied:
An example would have been if I needed a custom CurrentTraceContext that does some magic for non Ratpack things.  One case is the MDC addition but there are others such as some cross execution stuff between Ratpack and Cassandra.  In order to solve the Ratpack Execution -> Cassandra Execution hops I'm likely going to have to modify the CurrentTraceContext.  Honestly not sure how yet but I know its been a problem for us so I will be looking into it more soon.

In general though, the CurrentTraceContext is pluggable in Brave, so maintaining the "pluggability" in the Ratpack instrumentation would be ideal.

codefromthecrypt commented on src/test/groovy/ratpack/zipkin/ServerTracingModuleSpec.groovy:
```diff
@@ -265,8 +261,8 @@ class ServerTracingModuleSpec extends Specification {
 			Span span = reporter.getSpans().get(0)
-			span.binaryAnnotations.find { it -> it.key == TraceKeys.HTTP_STATUS_CODE } != null
-			span.binaryAnnotations.find { it -> it.key == Constants.ERROR } != null
+			span.tags().find { it -> it.key == TraceKeys.HTTP_STATUS_CODE } != null
```
might be a map assertion?

llinder replied:
Thanks for the suggestion!  Map assertions are awesome... I clearly don't use assertj enough :)

codefromthecrypt commented on src/test/groovy/ratpack/zipkin/internal/RatpackCurrentTraceContextSpec.groovy:
```diff
@@ -61,4 +62,23 @@ class RatpackCurrentTraceContextSpec extends Specification {
             traceContext == null
     }
 
+    def 'When TraceContext is null the context should be cleared'() {
```
thx

codefromthecrypt reviewed (approved):
LG!

llinder commented:
I've moved all of the CurrentTraceContext changes including the MDC parts into this PR https://github.com/hyleung/ratpack-zipkin/pull/41

I'm going to revise the Span V2 PR to build on that but only include V2 stuff.

llinder committed:
Upgrade to Span V2 model

llinder commented:
Ok I've updated this PR and build on top of https://github.com/hyleung/ratpack-zipkin/pull/41.  Once 41 is merged I will rebase this PR to remove the MDC commit.

hyleung commented:
@llinder I'm gonna try to do a little testing on this PR. Might take me a little while because I haven't been keeping up with Zipkin developments so need to catch up on the v2 stuff - but I haven't forgotten about this!

codefromthecrypt commented on gradle.properties:
```diff
@@ -1,4 +1,4 @@
 version=2.0.2-SNAPSHOT
 
-braveVersion=4.5.0
+braveVersion=4.7.1
```
4.8.0 :)

codefromthecrypt commented on src/main/java/ratpack/zipkin/ServerTracingModule.java:
```diff
@@ -33,8 +33,8 @@
-import zipkin.Span;
+import zipkin2.Endpoint;
+import zipkin2.Span;
 import zipkin.reporter.Reporter;
```
zipkin2.reporter.Reporter

hyleung replied:
[`Config#spanReporter`](https://github.com/llinder/ratpack-zipkin/blob/19a6724734d5a5f71967bb3f3795cff6f57bc9bd/src/main/java/ratpack/zipkin/ServerTracingModule.java#L118-L121)  needs to be updated to return `zipkin2.reporter.Reporter` too. cc/ @llinder 

[Edit] Well, I guess it wouldn't need to be updated since the class name would be the same.

hyleung replied:
I suppose this would mean that v1 `Reporter`s aren't going to be usable anymore, which would make this a breaking change.

codefromthecrypt commented on src/main/java/ratpack/zipkin/ServerTracingModule.java:
```diff
@@ -88,7 +88,7 @@ private static Endpoint buildLocalEndpoint(String serviceName, int port, @Nullab
   public static class Config {
     private String serviceName = "unknown";
-    private Reporter<Span> spanReporter = Reporter.NOOP;
+    private Reporter<Span> spanReporter = (Span s) -> {};
```
Reporter.NOOP is here now

codefromthecrypt commented on src/test/groovy/ratpack/zipkin/ServerTracingModuleSpec.groovy:
```diff
@@ -55,7 +53,7 @@ class ServerTracingModuleSpec extends Specification {
 								.clientSampler(HttpSampler.TRACE_ID)
 								.serverSampler(HttpSampler.TRACE_ID)
-								.spanReporter(Reporter.NOOP)
+								.spanReporter({Span s -> })
```
zipkin2.reporter.Reporter.NOOP is back

codefromthecrypt commented on src/test/groovy/ratpack/zipkin/support/TestReporter.groovy:
```diff
@@ -1,6 +1,6 @@
 
-import zipkin.Span
+import zipkin2.Span
 import zipkin.reporter.Reporter
```
zipkin2.reporter.Reporter

codefromthecrypt reviewed (commented):
made some notes now that we have a v2 native reporter (you can drop any dep on io.zipkin.reporter:zipkin-reporter for io.zipkin.reporter2:zipkin-reporter)

codefromthecrypt commented:
Well you can also support reportet v1 api (via tracing.builder.reporter) or
v1 format (via encoder)
https://github.com/openzipkin/brave/blob/master/brave/README.md#zipkin-v1-setup

On 24 Sep 2017 11:32, "Ho Yan Leung" <notifications@github.com> wrote:

> *@hyleung* commented on this pull request.
> ------------------------------
>
> In src/main/java/ratpack/zipkin/ServerTracingModule.java
> <https://github.com/hyleung/ratpack-zipkin/pull/40#discussion_r140646341>:
>
> > @@ -33,8 +33,8 @@
>  import ratpack.zipkin.internal.DefaultServerTracingHandler;
>  import ratpack.zipkin.internal.RatpackCurrentTraceContext;
>  import ratpack.zipkin.internal.ZipkinHttpClientImpl;
> -import zipkin.Endpoint;
> -import zipkin.Span;
> +import zipkin2.Endpoint;
> +import zipkin2.Span;
>  import zipkin.reporter.Reporter;
>
> I suppose this would mean that v1 Reporters aren't going to be usable
> anymore, which would make this a breaking change.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/hyleung/ratpack-zipkin/pull/40#discussion_r140646341>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAD61yK_fEatwn4Gryk_G950cB_jc2Kxks5slc1hgaJpZM4PYa45>
> .
>

codefromthecrypt commented:
Basically for compat I would recommend accepting a v1 reporter but
preferring v2 internally. You can convert a v1 reporter to a v2 one (ex
look at Tracing.Builder.reporter for a hint)

Commit:
Updates to reporter2

codefromthecrypt commented:
ok I made some changes here which seem to work. feel free to
copy/paste instead of trying to merge my commit if it helps
https://github.com/llinder/ratpack-zipkin/pull/2

llinder commented:
I've been struggling to keep up on my own PR sorry :(  I this covers the feedback and also incorporates @adriancole changes.  Looks like the code coverages is dropping so I will try to see what we are missing.

BTW I spent some time trying to get the ITHttpServer integration tests working.  It looks like I might need to do some refactoring on this to make it work with Ratpack.  Once I get a window of time again I will try to pick it up again and make  PR upstream when I get things to work locally.

hyleung commented:
@llinder no need to be sorry! From the [coverage report](https://codecov.io/gh/hyleung/ratpack-zipkin/src/6b40c4aa34735856f1699b59e99fd69d286d8243/src/main/java/ratpack/zipkin/ServerTracingModule.java#L122...140), it looks like it's a couple of missed code paths in the deprecated `spanReporter` method on the Config. The no-op case would probably the only test you could argue would be valuable. If we cover that case and the coverage check still fails, I'm cool with just letting it go.

codefromthecrypt commented:
thanks lance. you're a champ!

llinder commented:
All green!  I changed the legacy reporter handling a bit to push this up to the tracer itself.  The result should be the same, just slightly changed how the Tracing instance is built.

codefromthecrypt commented on src/main/java/ratpack/zipkin/ServerTracingModule.java:
```diff
@@ -59,14 +59,21 @@ protected void configure() {
+        .localServiceName(config.serviceName);
+
+    // for backwards comparability, prefer legacy reporter if defined otherwise use the
+    if (config.legacyReporter != null) {
```
Ps i dont remember how I coded before but I think I probably coded to avoid a check like this. This check requires zipkin 1 in the runtime classpath (vs just compile). It isnt obvious until after you try to remove the dep ;) 300 KiB might not matter to people and easy enough to rewire later if asked.

hyleung replied:
huh, so if we removed zipkin 1 from the runtime class path, do we get an exception or something?

codefromthecrypt reviewed (approved):
Good work!

hyleung commented:
@llinder I pulled the branch to do a quick test - pretty much just dropped the updated dependency in and everything worked w/o any other code changes, so in terms of legacy compatibility, we're good. Still need to get myself setup to test the v2 stuff.

codefromthecrypt commented:
yeah you'd get a classnotfound. In brave, we guard with an internal api
Platform.zipkin1Present I think. however, if the reporter is converted
prior to this step, then there's no chance of classnotfound

On Thu, Sep 28, 2017 at 11:30 AM, Ho Yan Leung <notifications@github.com>
wrote:

> *@hyleung* commented on this pull request.
> ------------------------------
>
> In src/main/java/ratpack/zipkin/ServerTracingModule.java
> <https://github.com/hyleung/ratpack-zipkin/pull/40#discussion_r141519670>:
>
> >          .sampler(config.sampler)
>          .currentTraceContext(new RatpackCurrentTraceContext())
>          .localEndpoint(buildLocalEndpoint(config.serviceName, serverConfig.getPort(),
>              serverConfig.getAddress()))
> -        .localServiceName(config.serviceName)
> -        .reporter(config.spanReporter)
> -        .build();
> +        .localServiceName(config.serviceName);
> +
> +    // for backwards comparability, prefer legacy reporter if defined otherwise use the
> +    if (config.legacyReporter != null) {
>
> huh, so if we removed zipkin 1 from the runtime class path, do we get an
> exception or something?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/hyleung/ratpack-zipkin/pull/40#discussion_r141519670>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAD61zIXBUhY_Won2HvJt_7sIFY4pZ5eks5smxLFgaJpZM4PYa45>
> .
>

hyleung commented:
@adriancole wow, i wouldn't have expected that by just looking at the code! I tried it just now and I somehow got a compile time error ¯\_(ツ)_/¯ . Anyhow, I'm not too bothered either way (if we leave the dependency in or out) - thoughts, @llinder?

codefromthecrypt commented:
>
> @adriancole <https://github.com/adriancole> wow, i wouldn't have expected
> that by just looking at the code! I tried it just now and I somehow got a
> compile time error ¯_(ツ)_/¯ . Anyhow, I'm not too bothered either way (if
> we leave the dependency in or out) - thoughts, @llinder
> <https://github.com/llinder>?
>
yeah basically it is splitting hairs a bit. ratpack-zipkin does need a
compile dep, but a user configuring ratpack-zipkin could go without it at
runtime, if the legacy span reporter wasn't required to be checked. Ex I
use a separate project to verify this

https://github.com/openzipkin/brave/tree/master/brave/src/it/nozipkin1

llinder commented:
Didn't even consider compile vs runtime dependency on Zipkin 1.  Definitely some merit in not making the dependency required.  I will revert this change and get code coverage up a different way :)

hyleung commented:
@llinder this'll need a rebase after #45. Are you still looking trying to make the v1 dependency not-required?

llinder commented:
@hyleung I added the code back that Adrian suggested so that should remove the hard v1 dependency https://github.com/hyleung/ratpack-zipkin/pull/40/files#diff-6d11129c36716a94c98d2e75a65b5a5cR154.  I've also upgrade Gradle wrapper to 2.12 so we can declare the v1 reporter as `compileOnly` so this shouldn't show up in the transitive dependencies now.

Also rebased against master so this should be good to go.

codefromthecrypt commented:
ps I will write release notes later, but propagationFactory is a big part of brave 4.9 going out now. Notably, this allows you to use amazon (single-header) propagation or even push other things

https://github.com/openzipkin/brave/tree/master/propagation/aws#brave-propagation-aws
https://github.com/openzipkin/brave/blob/master/brave/README.md#propagating-extra-fields

llinder commented:
Thanks for the heads up on 4.9.  I will look at updating this and adding the propagationFactory to the module config later today.

llinder commented:
Rebased and added `propagationFactory` support.  @hyleung what do you think about merge this?  Any additional gaps we should fill?

hyleung commented:
@llinder I'll take a quick look (hopefully) tonight - should be fine to merge unless there were a lot of changes required to add `propagationFactory`.

hyleung commented:
@llinder I pulled the branch and did a quick check to make sure that things work as far as backward compatibility goes. Still want to test the v2 spans but I'll need to futz about with my test app and setup a little before I can verify that. Is this something that's urgent for you or can I take a couple of days to do a few more checks?

hyleung commented:
Not sure if I just missed something really obvious, but I moved my test app over to `zipkin2` dependencies, but needed to add a dep with `zipkin.reporter.Reporter` for it to compile.

llinder commented:
@hyleung no this isn't urgent so please don't feel rushed.

As far as I know `zipkin.reporter.Reporter` dependency should only be required if your actually using the V1 reporter.  This was one of the changes I made a while ago, which removed the hard dependency on this.  If your test app uses that then it makes sense you had to add that for it to compile.  Most users going forward won't need this so it is only an optional dependency now.

hyleung commented:
@llinder ok, cool. As far as I know I'm not using the v1 reporter. I'll keep digging - might have to trouble you for a quick second set of 👀s. 

I haven't been keeping up to date with Zipkin developments, so I'm not quite sure how to go about getting set up to test v2 (I've been using the docker-compose stack).

codefromthecrypt commented:
Server exposes both endpoints so you are all good. Hopefully you have a
compile dep only building the project itself. End consumers should be able
to leave out the v1 dep but that is something to verify.

On 6 Nov 2017 13:18, "Ho Yan Leung" <notifications@github.com> wrote:

> @llinder <https://github.com/llinder> ok, cool. As far as I know I'm not
> using the v1 reporter. I'll keep digging - might have to trouble you for a
> quick second set of 👀s.
>
> I haven't been keeping up to date with Zipkin developments, so I'm not
> quite sure how to go about getting set up to test v2 (I've been using the
> docker-compose stack).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/hyleung/ratpack-zipkin/pull/40#issuecomment-342049000>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAD61yzC2s9oRDQotKsnZyPdEZ2QFd0Zks5szpakgaJpZM4PYa45>
> .
>

hyleung commented:
@adriancole hmm...i'm running the latest `docker-zipkin` but `curl localhost:9411/api/v2/services` gives me a `404`. I exec'd into the container so I'm assuming there's not some weird networking thing going on.

codefromthecrypt commented:
did you docker-compose pull? hit the /info endpoint to verify the version
is indeed latest?

On Tue, Nov 7, 2017 at 11:52 AM, Ho Yan Leung <notifications@github.com>
wrote:

> @adriancole <https://github.com/adriancole> hmm...i'm running the latest
> docker-zipkin but curl localhost:9411/api/v2/services gives me a 404. I
> exec'd into the container so I'm assuming there's not some weird networking
> thing going on.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/hyleung/ratpack-zipkin/pull/40#issuecomment-342368010>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAD61-L0gKf8FE_UEk8QNWGyuOgnU9Fcks5sz9P0gaJpZM4PYa45>
> .
>

hyleung commented:
@adriancole yeah, I did do a pull.

```
$  http "http://localhost:9411/info"
HTTP/1.1 200
Content-Type: application/vnd.spring-boot.actuator.v1+json;charset=UTF-8
Date: Wed, 08 Nov 2017 00:03:46 GMT
Transfer-Encoding: chunked
X-Application-Context: zipkin-server:shared:9411

{
    "zipkin": {
        "version": "2.2.1"
    }
}

$  http "http://localhost:9411/api/v2/services"
HTTP/1.1 404
Content-Length: 0
Date: Wed, 08 Nov 2017 00:04:27 GMT
X-Application-Context: zipkin-server:shared:9411
```

hyleung commented:
Here's the [branch](https://github.com/hyleung/ratpack-zipkin-example/tree/upgrade-to-v2) in my test project where I've been trying to upgrade to v2. When I *don't* include the v1 reporter dependency, I get a compilation error:

```
./gradlew clean check
:clean
:compileJava
/Users/hyleung/Projects/ratpack-zipkin-example/src/main/java/com/example/server/App.java:29: error: cannot access Reporter
                  .spanReporter(spanReporter());
                  ^
  class file for zipkin.reporter.Reporter not found
1 error
:compileJava FAILED
```

codefromthecrypt commented:
Thx will look into it and release a patch if it isnt a mutual eyes crossing
problem!

On 8 Nov 2017 8:05 am, "Ho Yan Leung" <notifications@github.com> wrote:

> @adriancole <https://github.com/adriancole> yeah, I did do a pull.
>
> $  http "http://localhost:9411/info"
> HTTP/1.1 200
> Content-Type: application/vnd.spring-boot.actuator.v1+json;charset=UTF-8
> Date: Wed, 08 Nov 2017 00:03:46 GMT
> Transfer-Encoding: chunked
> X-Application-Context: zipkin-server:shared:9411
>
> {
>     "zipkin": {
>         "version": "2.2.1"
>     }
> }
>
> $  http "http://localhost:9411/api/v2/services"
> HTTP/1.1 404
> Content-Length: 0
> Date: Wed, 08 Nov 2017 00:04:27 GMT
> X-Application-Context: zipkin-server:shared:9411
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/hyleung/ratpack-zipkin/pull/40#issuecomment-342665641>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAD615QYwHrZjASmsNK8KVz2eg2yJgVtks5s0PAzgaJpZM4PYa45>
> .
>

hyleung commented:
lol "mutual eyes crossing" - definitely stealing that!

Addition info, I'm running Docker for Mac:

Version 17.09.0-ce-mac34 (19605)
Channel: edge
c076099f52

I've also tried `docker-machine` with the same result. I've exec'd into the container so I'm assuming it isn't some weird networking issue. Looking at the logs as far as I can tell, the v2 handlers are being picked up, etc.

codefromthecrypt commented:
hmm I'm using curl and get this...

curl -v 192.168.99.100:9411/api/v2/services

*   Trying 192.168.99.100...

* Connected to 192.168.99.100 (127.0.0.1) port 9411 (#0)

> GET /api/v2/services HTTP/1.1

> Host: 192.168.99.100:9411

> User-Agent: curl/7.43.0

> Accept: */*

>

< HTTP/1.1 200

< X-Application-Context: zipkin-server:shared:9411

< Content-Type: application/json;charset=UTF-8

< Transfer-Encoding: chunked

< Vary: Accept-Encoding

< Date: Wed, 08 Nov 2017 06:51:02 GMT

<

* Connection #0 to host 192.168.99.100 left intact

[]

On Wed, Nov 8, 2017 at 1:29 PM, Ho Yan Leung <notifications@github.com>
wrote:

> lol "mutual eyes crossing" - definitely stealing that!
>
> Addition info, I'm running Docker for Mac:
>
> Version 17.09.0-ce-mac34 (19605)
> Channel: edge
> c076099f52
>
> I've also tried docker-machine with the same result. I've exec'd into the
> container so I'm assuming it isn't some weird networking issue. Looking at
> the logs as far as I can tell, the v2 handlers are being picked up, etc.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/hyleung/ratpack-zipkin/pull/40#issuecomment-342714839>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAD61yUwzmQhZReKMRF53_MbiDixHoF4ks5s0TwggaJpZM4PYa45>
> .
>

codefromthecrypt commented:
also just tried httpie

http 192.168.99.100:9411/api/v2/services

HTTP/1.1 200

Content-Encoding: gzip

Content-Type: application/json;charset=UTF-8

Date: Wed, 08 Nov 2017 06:54:50 GMT

Transfer-Encoding: chunked

Vary: Accept-Encoding

X-Application-Context: zipkin-server:shared:9411


[]

hyleung commented:
Hmm.... @llinder if you were able to verify that the v2 stuff works as expected (i.e. you're able to verify that traces are being recorded properly), I'm fine with that (I'll figure out what's up with my environment later). 

I'm still not able to get my demo project compiling w/o adding v1 dependencies (https://github.com/hyleung/ratpack-zipkin-example/tree/upgrade-to-v2) - if you've got the time, I'd really appreciate it if you could take a look to see if I'm just missing something really obvious.

codefromthecrypt commented:
there was a v1 encoding bug, can you update to brave 4.9.1 (which depends on zipkin2 2.2.2 which has the fix)? Sorry about that. was missing a test and backfilled a bunch

codefromthecrypt commented:
(ps this is unrelated to the compiling issue... sorry no coffee yet)

hyleung commented:
@adriancole This PR already upgrades to Brave 4.9.1 so think we're ok. I've also updated my `docker-zipkin` so that it's running 2.2.2. For whatever reason I'm still getting that 404 on `/services` but the traces seem to be recording fine (at least with OkHttp) ¯\_(ツ)_/¯.

Re. the compilation issue, still need to include `io.zipkin.reporter:zipkin-reporter:1.1.2` tho'.

codefromthecrypt commented:
can you give me the cli input to build in such a way I can repro the sample
problem?

I did ./gradlew clean build and it is angry about "Cannot perform signing
task ':signArchives' because it has no configured signatory"

On Sun, Nov 12, 2017 at 8:13 AM, Ho Yan Leung <notifications@github.com>
wrote:

> @adriancole <https://github.com/adriancole> This PR already upgrades to
> Brave 4.9.1 so think we're ok. I've also updated my docker-zipkin so that
> it's running 2.2.2. For whatever reason I'm still getting that 404 on
> /services but the traces seem to be recording fine (at least with OkHttp)
> ¯_(ツ)_/¯.
>
> Re. the compilation issue, still need to include
> io.zipkin.reporter:zipkin-reporter:1.1.2 tho'.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/hyleung/ratpack-zipkin/pull/40#issuecomment-343703979>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAD616R3nrYTa3aBxGbauRhXEjkobCI0ks5s1jgugaJpZM4PYa45>
> .
>

hyleung commented:
@adriancole argh 🤦‍♂️ I'm sorry - totally forgot about that, Ive been meaning to fix that. Need to update `build.gradle` so that it doesn't try to sign local builds. My bad 😞

llinder commented:
@hyleung I think I see the issue with the compile errors when the `zipkin-reporter:1.1.2` dependency isn't included.  It appears that we need to change the method name for `spanReporter` to something else so the compiler can disambiguate the argument types.  Simply changing it to `spanReporterV2` does the trick but is a bit ugly.  If you have a suggestion on the naming for this let me know.  For now I will just update my PR to use `spanReporterV2` for lack of a better name.

llinder committed:
Changed spanReporter method name to spanReporterV2 to avoid dependency on V1 jar

llinder commented:
@hyleung I've tested the name change to verify that the `zipkin-reporter:1.1.2` dependency is no longer required.  I've also tested the `ratpack-zipkin-example` project against a local running Zipkin 2.3.0 server and spans are reported correctly.

![image](https://user-images.githubusercontent.com/99359/32701955-8e274596-c7a4-11e7-9427-7b81737c72d7.png)

I'm fairly confident that things are working as expected now.  If you feel we should check anything else let me know and I will jump on it as quick as I can.

llinder committed:
Bump Brave dependency to 4.9.2

hyleung commented:
@llinder thanks! Looks good to me. Think we can merge it now :)

I'll update the README separately before doing a release (assuming we're in no particular hurry).

Re. the `spanReporterV2` naming, I think it's probably fine. And actually I was wondering how long we would be supporting both v1 and v2 reporters. Until it becomes really painful to support both, I figure it's probably more user-friendly to keep supporting both for a while.  We could even add a `spanReporterV1`, which can supersede the deprecated `spanReporter` so that at least we're being very explicit about which version is expected.

codefromthecrypt commented:
Agree on method name problem. In brave reporter was v1 so we made v2
spanReporter for this reason. Drat should have noticed!

On 13 Nov 2017 4:41 am, "Ho Yan Leung" <notifications@github.com> wrote:

Merged #40 <https://github.com/hyleung/ratpack-zipkin/pull/40>.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/hyleung/ratpack-zipkin/pull/40#event-1337747830>, or mute
the thread
<https://github.com/notifications/unsubscribe-auth/AAD61wwNC0NtfnALanTJMbN4WaDHoB4Kks5s11fogaJpZM4PYa45>
.
